debezium-connector-binlog/src/main/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSource.java

```diff
@@ -253,7 +253,18 @@ public void execute(ChangeEventSourceContext context, P partition, O offsetConte
                 // We've not yet seen any GTIDs, so that means we have to start reading the binlog from the beginning ...
                 client.setBinlogFilename(effectiveOffsetContext.getSource().binlogFilename());
                 client.setBinlogPosition(effectiveOffsetContext.getSource().binlogPosition());
-                initializeGtidSet("");
+                if (purgedServerGtidSet.isEmpty()) {
```

Comment:
Looking at `io.debezium.connector.mysql.jdbc.MySqlConnection.purgedGtidSet()` it seems to me that `null` is valid return value. Would the code be safe in that case?

Reply:
I don't think we can ever get `null` from [here](https://github.com/debezium/debezium/blob/main/debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/jdbc/MySqlConnection.java#L108-L110) (and same for [MariaDB](https://github.com/debezium/debezium/blob/main/debezium-connector-mariadb/src/main/java/io/debezium/connector/mariadb/jdbc/MariaDbConnection.java#L71))

Reply:
Just noted the logging message is misleading and needs to be fixed, so the test will run again anyway - adding also the check for `null` as it never hurts.